### PR TITLE
test: Kubeflow Pipelines V2 integration Tests

### PR DESCRIPTION
**Pull request by diegolovison**
**Description of your changes:**
Create a Kubeflow Pipelines V2 integration Testes using GitHub Action

Fixes: https://github.com/kubeflow/pipelines/issues/11052

**Checklist:**
- [ X ] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: .github/workflows/kubeflow-pipelines-integration-v2.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+name: Kubeflow Pipelines V2 integration Testes
```
Comment (DharmitD) [inline .github/workflows/kubeflow-pipelines-integration-v2.yml:1]: small typo, should be *Tests :)

File: backend/src/v2/test/requirements.txt [hunk 2]
```diff
@@ -1,9 +1,3 @@
 # install kfp sdk from local path
 -e ../../../../sdk/python
-# TODO(chensun): remove the deprecated dependencies once migrated tests.
--r ../../../../sdk/python/requirements-deprecated.txt
-ml-metadata==1.14.0
-minio==7.0.4
-google-cloud-storage
-fire
--e ../../../../samples/test/utils
+fire
```
Comment (DharmitD) [inline backend/src/v2/test/requirements.txt:3]: make sure to fix EoF here :)

File: backend/src/v2/test/Makefile [hunk 3]
```diff
@@ -9,19 +9,9 @@ include $(ENV_PATH)
...
-		--timeout_mins 60 \
-		--context $(GCS_ROOT)/src/context.tar.gz \
-		--gcs_root $(GCS_ROOT)/data \
-		--gcr_root $(GCR_ROOT) \
-		--kfp_package_path "$(KFP_PACKAGE_PATH)"
-
-.PHONY: upload
-upload: context
-	python -u scripts/upload_gcs_blob.py tmp/context.tar.gz $(GCS_ROOT)/src/context.tar.gz
+		&& python -u ../../../../samples/v2/sample_test.py
```
Comment (gmfrasca) [inline backend/src/v2/test/Makefile:14]: going off the diff here, can you not just leave this line alone (ie `python -u sample_test.py`)?  It appears to already be in that directory as sample_test.py only exists there

Comment (diegolovison) [reply backend/src/v2/test/Makefile:14]: It was changed because the path was wrong.

Comment (gmfrasca) [reply backend/src/v2/test/Makefile:14]: updated: it appears the 'old' command may not be pointed correctly, or the `upload` Target run before the command somehow changes the working directory to a place where the script does exist.  tl;dr: not a problem

File: .github/workflows/kubeflow-pipelines-integration-v2.yml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    paths:
+      - '.github/workflows/kubeflow-pipelines-integration-v2.yml'
+      - 'samples'
+      - 'core'
+      - 'backend'
```
Comment (hbelmiro) [inline .github/workflows/kubeflow-pipelines-integration-v2.yml:12]: `backend` is not included in https://github.com/GoogleCloudPlatform/oss-test-infra/blob/68230effbdc9d59429b205475325c3212ac887d7/prow/prowjobs/kubeflow/pipelines/kubeflow-pipelines-presubmits.yaml#L61.
Do you really think it should be included here?

File: backend/src/v2/test/Makefile [hunk 5]
```diff
@@ -9,19 +9,9 @@ include $(ENV_PATH)
...
 
 .PHONY: context
 context:
```
Comment (hbelmiro) [inline backend/src/v2/test/Makefile:17]: This file adds nothing to the logic and can be completely removed. You can invoke `sample_test.py` from `backend/src/v2/test/integration-test.sh`.

**Review by hbelmiro — COMMENTED** [review]
Looks good in general. I just left two small comments.
Can you also add the link for the removal of https://github.com/GoogleCloudPlatform/oss-test-infra/blob/68230effbdc9d59429b205475325c3212ac887d7/prow/prowjobs/kubeflow/pipelines/kubeflow-pipelines-presubmits.yaml#L60 to the description of this PR?

File: .github/workflows/kubeflow-pipelines-integration-v2.yml (hunk 4 above)
Comment (diegolovison) [reply .github/workflows/kubeflow-pipelines-integration-v2.yml:12]: Sure!

File: backend/src/v2/test/Makefile (hunk 5 above)
Comment (diegolovison) [reply backend/src/v2/test/Makefile:17]: Well, I decided not to change the scope of things. Those files probably are going to change again once we starting moving to our CI. For now, I kept as closer to the original. Any problem?

Comment (hbelmiro) [reply backend/src/v2/test/Makefile:17]: I would remove it anyway. But feel free to decide it.

**Review by chensun — APPROVED** [review]

**Pull request merged**